internal/cbor/base.go

```diff
@@ -1,5 +1,13 @@
 package cbor
 
+// JSONMarshaler knows how to marshal an interface to JSON encoded byte slice.
+// Making it package level instead of embedded in Encoder brings
+// some extra efforts at importing, but avoids value copy when
+// the functions of Encoder being invoked.
+// DO REMEMBER to set this variable at importing, or
+// you might get a nil pointer dereference panic at runtime.
+var JSONMarshaler func(v interface{}) ([]byte, error)
```

Comment:
Why nor init this as `var JSONMarshFunc = json.Marshal`?

Reply:
Because it's an internal package, designed for non-independent usage. I'd like to keep the setting explicitly to avoid someone happens to left out the configuration. Also I think it makes the configuration clear and global once, no confusion will be made with the default marshal func.